frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -37,4 +45,129 @@ public void annotate(PsiElement element, AnnotationHolder holder) {
           .range(branch).create();
     }
   }
+
+  private void processIndentationElement(PsiElement element, AnnotationHolder holder) {
+    PsiElement parent = element.getParent();
+    assert parent != null : "Element has not parent";
+
+    if (parent instanceof MacheteFile) {
+      return;
+    }
+
+    var prevMacheteGeneratedEntryOption = getPrevSiblingMacheteGeneratedEntry(parent);
+    if (prevMacheteGeneratedEntryOption.isEmpty()) {
+      holder.newAnnotation(HighlightSeverity.ERROR, "First entry cannot be indented").range(element).create();
+      return;
+    }
+
+    int prevLevel;
+    int thisLevel;
+    boolean hasPrevLevelCorrectWidth;
+
+    IndentationParameters indentationParameters = findIndentationParameters(element);
+
+    var prevIndentationNodeOption = getIndentationNodeFromMacheteGeneratedEntry(prevMacheteGeneratedEntryOption.get());
+    if (prevIndentationNodeOption.isEmpty()) {
+      prevLevel = 0;
+      hasPrevLevelCorrectWidth = true;
+    } else {
+      var prevIndentationText = prevIndentationNodeOption.get().getText();
+      hasPrevLevelCorrectWidth = prevIndentationText.length() % indentationParameters.indentationWidth == 0;
+      prevLevel = prevIndentationText.length() / indentationParameters.indentationWidth;
+    }
+
+    var thisIndentationText = element.getText();
+
+    if (thisIndentationText.chars().filter(c -> c != indentationParameters.indentationCharacter).count() > 0) {
+      holder.newAnnotation(HighlightSeverity.ERROR, "Indentation character does not match the first indented line")
+          .range(element).create();
+      return;
+    }
+
+    if (thisIndentationText.length() % indentationParameters.indentationWidth != 0) {
+      holder.newAnnotation(HighlightSeverity.ERROR,
+          "Indentation width is not multiple of ${indentationParameters.indentationWidth}" +
+              " as first indented line suggests")
+          .range(element).create();
+    }
+
+    thisLevel = thisIndentationText.length() / indentationParameters.indentationWidth;
+
+    if (hasPrevLevelCorrectWidth && thisLevel > prevLevel + 1) {
+      holder.newAnnotation(HighlightSeverity.ERROR, "Indentation level does not math")
```

Comment:
`match`

Reply:
Maybe even for super-clarity sth like:
```suggestion
      holder.newAnnotation(HighlightSeverity.ERROR, "Too much indent on this line")
```